tests/pthread/main_thread_join.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright 2019 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+#include <assert.h>
+#include <emscripten.h>
+#include <pthread.h>
+#include <stdio.h>
+
+pthread_t thread;
+
+void loop() {
+  void* retval;
+  if (pthread_tryjoin_np(thread, &retval) == 0) {
```

Comment:
might be worth testing that tryjoin works with and without PTHREAD_POOL and a case where it fails to join. could be a future PR

Reply:
Sounds good, added now.